server/neptune/workflows/internal/activities/execute.go

```diff
@@ -1,12 +1,90 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
 
-type executeCommandActivities struct{}
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime/cache"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/steps"
+)
+
+type executeCommandActivities struct {
+	versionCache cache.ExecutionVersionCache
+
+	DefaultTFVersion *version.Version
+
+	// TerraformBinDir is the directory where Atlantis downloads Terraform binaries.
+	TerraformBinDir string
+}
+
+type ExecuteCommandResponse struct {
+	Output string
+	Error  error
+}
 
 type ExecuteCommandRequest struct {
+	Step          steps.Step
+	CustomEnvVars map[string]string
+	Envs          map[string]string
+
+	Path string
+	// TerraformVersion is the version of terraform we should use when executing
+	// commands for this project. This can be set to nil in which case we will
+	// use the default Atlantis terraform version.
+	TerraformVersion *version.Version
 }
 
-func (t *terraformActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) error {
-	return nil
+func (t *executeCommandActivities) ExecuteCommand(ctx context.Context, request ExecuteCommandRequest) ExecuteCommandResponse {
+
+	terraformVersion := t.DefaultTFVersion
+	if request.TerraformVersion != nil {
+		terraformVersion = request.TerraformVersion
+	}
+
+	// Ensure version exists
+	_, err := t.versionCache.Get(terraformVersion)
+	if err != nil {
+		return ExecuteCommandResponse{
+			Output: "",
+			Error:  errors.Wrapf(err, "getting version %s", request.TerraformVersion),
+		}
+	}
+
+	terraformEnvVars := map[string]string{
+		"ATLANTIS_TERRAFORM_VERSION": terraformVersion.String(),
+		"PATH":                       fmt.Sprintf("%s:%s", os.Getenv("PATH"), t.TerraformBinDir),
+	}
```

Comment:
What's the point of this if we're just running the command specified in the step?

Reply:
Looking at the [docs](https://www.runatlantis.io/docs/custom-workflows.html), we execute run steps with some environment variables pre-configured right? I just copied over the environment variables defined in our [run_step_runner](https://github.com/lyft/atlantis/blob/8fb410ff66f94268557ddf30c779f89b2710d464/server/core/runtime/run_step_runner.go#L40-L60) and modified it to support operations off the base branch only.

Reply:
yeah we shouldn't just copy things over if it's not valuable for our usecase.  In this case why do we care about terraform versions for custom run steps?